### ISSUE TITLE
Merge Concept(s) is not working

### ISSUE THREAD

**Issue by singhik** [NCIEVS/nci-protege5#462]
@bdionne : Hi, Merge of Concept is not working, the Surviving Concept does not get the Properties of Retiring Concept, post Merge Completion. 

Attached document has the detailed screen prints .

[Merge_Issue.docx](https://github.com/NCIEVS/nci-protege5/files/5036980/Merge_Issue.docx)

**Comment by bdionne** [comment]
@singhik - It's difficult to see the steps to reproduce this. The screenshots aren't really complete. Which class was created first? Was one a copy of the other or is the name test1-copy a conincidence. 

Seems to work fine here

<img width="1022" alt="Screen Shot 2020-08-06 at 2 26 32 PM" src="https://user-images.githubusercontent.com/31741/89568168-e8700600-d7f0-11ea-9633-0a51732e35f0.png">

**Comment by singhik** [comment]
@bdionne : Steps to Reproduce are in the attached document, Test1 was created first .Test1-Copy is a naming convention .

**Comment by bdionne** [comment]
@singhik - well I tried to guess the order of the steps from my screenshots, and it works fine for me

I'll keep guessing at how you may have done this and see if I can reproduce it

**Comment by singhik** [comment]
@bdionne : Hi Bob, i will re-do it from scratch and let you know, will send a new document if the issue persists, you can keep it on hold till then

**Comment by bdionne** [comment]
@singhik - a list of steps is preferrable, easier to follow and replicate, and easier to search in the future. All these word docs have zero future value

**Comment by singhik** [comment]
@bdionne : not really, screen print documents provide the operation followed, and for Merge it has them , Merge is obviously between concept and it is apparent that Concept can be merged only after creation ,so the pre-requisite of a concept create is but obvious . Attached document had Merge operation screen prints listing all the properties of individual concepts and the one merged and retired .

**Comment by singhik** [comment]
@bdionne : Hi Bob, please find below  the detailed steps  with screen prints:: 

1. Create a concept  f.e. 'Test_1' , per below : 

![image](https://user-images.githubusercontent.com/36314371/89594763-b70f2e80-d820-11ea-9656-adbc10943d51.png)

2. Create a concept f.r. 'Test_2' , per below : 

![image](https://user-images.githubusercontent.com/36314371/89594833-ee7ddb00-d820-11ea-8f8b-a09d0df66f31.png)

3. Peform  a complex edit->Merge , with 'Test_1' as Surviving Concept and 'Test_2' as Retiring Concept, per below : 

![image](https://user-images.githubusercontent.com/36314371/89594917-2edd5900-d821-11ea-93a8-9fbaa63841cf.png)

![image](https://user-images.githubusercontent.com/36314371/89594934-369cfd80-d821-11ea-886a-a2858b7afb94.png)

![image](https://user-images.githubusercontent.com/36314371/89594957-3ef53880-d821-11ea-9354-96f8cbf0ac23.png)

4. 'Test_2' is Retired as Expected : 

![image](https://user-images.githubusercontent.com/36314371/89594977-53d1cc00-d821-11ea-930a-6a48039b8196.png)

5. 'Test_1"  does **not** show the properties of 'Test_2' post Merge ?? Which i would need your input for :: 

![image](https://user-images.githubusercontent.com/36314371/89595103-a90ddd80-d821-11ea-9857-703153a25b16.png)

**Comment by bdionne** [comment]
@singhik - this does not reproduce for me. I tried using `SmallBase` which worked fine, and then I loaded up a copy of production thesaurus I have from June, and that worked fine also. I notice that you seem to have some content that is different so I followed it as close as I can. Here is the steps I executed:

 1. create test1. Add additional definition to it with source `SPL`, additional `FULL_SYN` with source `SY`, and add an `ALT-DEF`with source `PQCMC`

 2. create test2. Use `CBDD` for the source of the `DEFINITION`. Add an additional `ALT-DEF` with source `NCI`

 3. merge using the normal process. Logged in as manager so classes immediately merge

Here's what I got, both the `DEFINITION` and the `ALT-DEF` were copied fine. Perhaps this is a refresh issue or something related to `.fileprefs`

<img width="1191" alt="Screen Shot 2020-08-07 at 5 39 25 AM" src="https://user-images.githubusercontent.com/31741/89633297-9f13cb00-d871-11ea-8efd-60c3f86ced6a.png">

**Comment by bdionne** [comment]
I see why it doesn't copy. In class `Test_2`, the values for the `DEFINITION` and `ALT_DEFINITION` annotations are equal to the label on the class, so they are not copied. I'm not sure why that check is there, probably because in our case the label is usually the `rdfs:label` which is the same as the `Preferred_Name` and the value for the `NCI/PT` `FULL_SYN`.

None of those are copied.

**Comment by fragosog** [comment]
I don't know why that check is in there either, and it's not there in the business rules section of the wiki.  In reality something like this wouldn't happen, e.g. have a concept "therapeutic agent" have a text definition = "therapeutic agent".  But if we had such a business rule, it would certainly fire during the creation of the definition.  I'd suggest to keep the merge processing as is for this release and maybe eliminate the check in a future release.

**Comment by bdionne** [comment]
what about `FULL_SYNS`? Do we want to allow multiple `FULL_SYN` values that are identical to the `NCI/PT` value? Duplicates essentially but with potentially different term sources and groups?

**Comment by bdionne** [comment]
see #463

**Comment by singhik** [comment]
From @NCIEVS :: 
"No, we can close #462, it only fails on an edge case that shouldn't exist in production.  if we need to follow up on one of the aspects that were raised it would be in a different ticket."

**Comment by bdionne** [comment]
so this is both and edge case, and something we don't want to let the modelers do.

**Comment by fragosog** [comment]
> so this is both and edge case, and something we don't want to let the modelers do.
Yes on both

> what about FULL_SYNS? Do we want to allow multiple FULL_SYN values that are identical to the 
> NCI/PT value? Duplicates essentially but with potentially different term sources and groups?
Yes, we want to allow the dups that have a different source or group
